File: Periodic Practice1/startup.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    int c[4];
    for (int i = 0; i < 4; i++) {
        cin >> c[i];
    }

    int eaten[4]={0};
    int pos[4]={0 , 1 , 2 ,3};
    int person=0;

    while(true){
        int section=pos[person];

        c[section]--;
        eaten[person]++;

        if(c[section]==0){
            break;
        }

        for(int i = 0; i < 4; i++){
            pos[i]= (pos[i]+1)%4;
        }
        person= (person+1)%4;
    }
    for (int i = 0; i < 4; i++) {
        cout << eaten[i] << " ";
    }
}

File: Periodic Practice1/startup.py
c=list(map(int , input().split()))

pos=[0,1,2,3]
eaten=[0,0,0,0]
person=0

while True:

    section=pos[person]
    c[section]-=1
    eaten[person]+=1

    if c[section]==0: 
        break

    for i in range(4):
        pos[i]=(pos[i]+1)%4

    person=(person+1)%4

for i in range(4):
    print(eaten[i])

